src/main/ruby/core/truffle/interop.rb

```diff
@@ -65,6 +65,44 @@ def self.enumerable(foreign)
     class Foreign
 
     end
+    
+    class ObjectLiteral
+    
+      def method_missing(sent_name, *args)
+        sent_name_s = sent_name.to_s
+        if sent_name_s.end_with?('=')
+          name = sent_name_s[0..-2].to_sym
+        else
+          name = sent_name
+        end
+        Truffle.privately { singleton_class.attr_accessor name }
```

Comment:
Do we need this method_missing magic here?
Why not simply use instance_variable_get/instance_variable_set in []/[]=?
That would compile better and avoid deopts, no?